Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorWindow.cs

```diff
@@ -260,34 +271,48 @@ private void OnLostFocus()
             #endif
         }
 
-        private void OnDestroy()
-        {
-            ConfirmSaveChangesIfNeeded();
-        }
-
-        private void ConfirmSaveChangesIfNeeded()
+        private void HandleOnDestroy()
         {
-            // Do we have unsaved changes?
+            // Do we have unsaved changes that we need to ask the user to save or discard?
             if (!m_IsDirty)
                 return;
 
-            var result = InputActionsEditorWindowUtils.ConfirmSaveChanges(m_AssetPath);
+            // Get target asset path from GUID, if this fails file no longer exists and we need to abort.
+            var assetPath = AssetDatabase.GUIDToAssetPath(m_AssetGUID);
+            if (string.IsNullOrEmpty(assetPath))
+                return;
+
+            // Prompt user with a dialog
+            var result = Dialog.InputActionAsset.ShowSaveChanges(assetPath);
             switch (result)
             {
-                case InputActionsEditorWindowUtils.ConfirmSaveChangesDialogResult.Save:
+                case Dialog.Result.Save:
                     Save();
                     break;
-                case InputActionsEditorWindowUtils.ConfirmSaveChangesDialogResult.Cancel:
+                case Dialog.Result.Cancel:
                     // Cancel editor quit. (open new editor window with the edited asset)
                     ReshowEditorWindowWithUnsavedChanges();
                     break;
-                case InputActionsEditorWindowUtils.ConfirmSaveChangesDialogResult.DontSave:
+                case Dialog.Result.Discard:
                     // Don't save, quit - reload the old asset from the json to prevent the asset from being dirtied
-                    AssetDatabase.ImportAsset(m_AssetPath);
+                    // TODO I doubt this is needed since we were operating on a copy?!
```

Comment:
Delete this then ?

Reply:
Removed in https://github.com/Unity-Technologies/InputSystem/pull/1850/commits/a8c93599cbc79b3f13fa0a0cad68d20483954d32